aztec/src/main/kotlin/org/wordpress/aztec/AztecText.kt

```diff
@@ -345,100 +346,125 @@ class AztecText : EditText, TextWatcher {
     }
 
-    protected fun bulletValid() {
-        val lines = TextUtils.split(editableText.toString(), "\n")
+    private fun bulletValid() {
+        bulletValid(selectionStart, selectionEnd)
+    }
 
-        for (i in lines.indices) {
-            if (containBullet(i)) {
-                continue
-            }
+    private fun bulletValid(start: Int, end: Int) {
+        if (start != end) {
+            val selectedText = editableText.substring(start + 1..end - 1)
 
-            var lineStart = 0
-            for (j in 0..i - 1) {
-                lineStart = lineStart + lines[j].length + 1 // \n
-            }
+            //multiline text selected
+            if (selectedText.indexOf("\n") != -1) {
+                val indexOfFirstLineBreak = editableText.indexOf("\n", end)
+
+                val endOfList = if (indexOfFirstLineBreak != -1) indexOfFirstLineBreak else editableText.length
+                val startOfList = editableText.lastIndexOf("\n", start)
+
+                val selectedLines = editableText.subSequence(startOfList + 1..endOfList - 1) as Editable
+
+                var numberOfLinesWithSpanApplied = 0
+                var numberOfLines = 0
+
+                val lines = TextUtils.split(selectedLines.toString(), "\n")
+
+                for (i in lines.indices) {
+                    numberOfLines++
+                    if (containBullet(i, selectedLines)) {
+                        numberOfLinesWithSpanApplied++
+                    }
+                }
+
+                if (numberOfLines == numberOfLinesWithSpanApplied) {
+                    bulletInvalid()
+                } else {
+                    editableText.setSpan(AztecBulletSpan(bulletColor, bulletRadius, bulletGapWidth), startOfList + 1, endOfList, Spanned.SPAN_EXCLUSIVE_INCLUSIVE)
+                }
 
-            val lineEnd = lineStart + lines[i].length
-            if (lineStart >= lineEnd) {
-                continue
-            }
 
-            // Find selection area inside
-            var bulletStart = 0
-            var bulletEnd = 0
-            if (lineStart <= selectionStart && selectionEnd <= lineEnd) {
-                bulletStart = lineStart
-                bulletEnd = lineEnd
-            } else if (selectionStart <= lineStart && lineEnd <= selectionEnd) {
-                bulletStart = lineStart
-                bulletEnd = lineEnd
             }
 
-            if (bulletStart < bulletEnd) {
-                editableText.setSpan(AztecBulletSpan(bulletColor, bulletRadius, bulletGapWidth), bulletStart, bulletEnd, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
+        } else {
+            val boundsOfSelectedText = getSelectedTextBounds(editableText, start, end)
+
+            val startOfLine = boundsOfSelectedText.start
+            var endOfLine = boundsOfSelectedText.endInclusive
+
+            if (startOfLine == endOfLine) {   //line is empty
+                consumeEditEvent = true
+                editableText.insert(startOfLine, "\u200B")
```

Comment:
I've used the same special character to aid parsing of `<div>`s [here](https://github.com/wordpress-mobile/WordPress-Aztec-Android/blob/develop/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L43). Couldn't that cause some conflicts for you?